### ISSUE TITLE
Widescreen resolution and display scaling

### ISSUE THREAD

**Issue by lxkurko** [openlierox/openlierox#799]
Two ideas to improve game experience on modern systems as 16:9 displays are the standard now:

1) Widescreen resolution: When playing in fullscreen mode on a widescreen display, 640x480 either leaves large black bars on the screen or gets stretched and ugly. My suggestion is to change the base size of the game window to 720x405 pixels. This way, the levels do not have to be modified. You see a bit more on the horizontal line and slightly less on the vertical line, but not much. Also, since 0.59 already extends the visible area compared to 0.58, the vertical change compared to 0.58 will be almost negligible.

Another possible resolutions might be 640x360 or 960x540 if integer multiplies of 1920x1080 are desired before scaling.

2) Resolution scaling: Currently, if you switch to fullscreen, OLX will change display resolution. It takes some time as the display hardware adjusts itself, and often causes blurred graphics when the display (LCD) operates on non-native resolution. So if possible without horribly excessive work, do upscaling to screen resolution in OLX. Also, allowing window scaling would be useful if someone wants to play OLX in windowed mode on a high-resolution display. Modern computers should be fast enough so that performance does not become a problem.

Any estimate about the work required to do these? I could try it myself too.

**Comment by Ghostbird** [comment]
Actually, when I build the Github source, it already works. This is great!

**Comment by lxkurko** [comment]
Yep, 0.59 seems to scale to fullscreen nicely. 0.58 wants to change the display resolution.

I did some experiments and widescreen at 720x405 seems to work quite nicely. It would just require redesigning of the interface and that's quite a lot of work.

**Comment by albertz** [comment]
Implemented on current master.
Fullscreen now uses `SDL_WINDOW_FULLSCREEN_DESKTOP`
and renders through `SDL_RenderSetLogicalSize`
with `SDL_HINT_RENDER_SCALE_QUALITY = "best"` (src/client/AuxLib.cpp),
so the game scales to the display instead of switching the monitor resolution.
Widescreen for local games was added in #927 (7be0fcd93):
the render width follows the desktop aspect ratio,
while network games stay 640-wide so a wide screen gives no advantage.
You already confirmed the scaling works from a git build. Closing as done.
